pennylane/optimize/qng.py

```diff
@@ -73,68 +73,35 @@ class QNGOptimizer(GradientDescentOptimizer):
         "Quantum Natural Gradient."
         `Quantum 4, 269 <https://doi.org/10.22331/q-2020-05-25-269>`_, 2020.
 
-    .. note::
-
-        The QNG optimizer supports single QNodes or :class:`~.ExpvalCost` objects as objective functions.
-        Alternatively, the metric tensor can directly be provided to the :func:`step` method of the optimizer,
-        using the ``metric_tensor_fn`` argument.
-
-        For the following cases, providing metric_tensor_fn may be useful:
-
-        * For hybrid classical-quantum models, the "mixed geometry" of the model
-          makes it unclear which metric should be used for which parameter.
-          For example, parameters of quantum nodes are better suited to
-          one metric (such as the QNG), whereas others (e.g., parameters of classical nodes)
-          are likely better suited to another metric.
-
-        * For multi-QNode models, we don't know what geometry is appropriate
-          if a parameter is shared amongst several QNodes.
```

Comment:
@albi3ro I suggest keeping this part (up to line 91), with the following change:

- Changing the first sentence to say "The QNG optimizer by default only supports single QNodes as objective functions." (that is, remove mention of `ExpvalCost`).

- Maybe moving it after the example?

The reason is that a lot of users expect `QNGOptimizer` to work on arbitrary cost functions, but this is not the case, for theoretical reasons --- it is not clear how to generalize the theory beyond a single variational circuit. We used to get a lot of questions/bug reports about this, so decided to be explicit in the docstring.